### PR TITLE
Allow passing dictionary of http client parameters to DiscoveryV1 api constructor

### PR DESCRIPTION
**Pull request by breathe**
Fix for #327

### CODE REVIEW

**Comment by germanattanasio** [comment]
@jsstylos can you take a look at this? I think this is not a breaking change so we could add it to the generator.

**Comment by jsstylos** [comment]
This looks good, we will need to add it in the generator due to our new SDK generation process, but that also means we will be able to add it to all of the services, not just Discovery.

File: watson_developer_cloud/discovery_v1.py [hunk 1]
```diff
@@ -43,7 +43,7 @@ class DiscoveryV1(WatsonService):
     VERSION_DATE_2017_06_25 = '2017-06-25'
     VERSION_DATE_2016_12_01 = '2016-12-01'
 
-    def __init__(self, version, url=default_url, username=None, password=None):
+    def __init__(self, version, url=default_url, username=None, password=None, http_config={}):
```
Comment (mkistler) [inline watson_developer_cloud/discovery_v1.py]: I'd like to suggest an alternate solution -- just add a method to the `watson_service` class for setting the http_config.  This is probably needed anyway, and once we have it, we don't really need to add an `http_config` parameter to all the service constructors, or change the generator -- it's a much simpler solution.

Comment (jsstylos) [reply watson_developer_cloud/discovery_v1.py]: That sounds like a good idea.

**Pull request closed without merging**